Refresh Visual Studio Edge debugging docs (#1360)
MicrosoftDocs/edge-developer: microsoft-edge/visual-studio/index.md

### Create a new ASP.NET Core web app  

Open Visual Studio 2019 and select **Create a new project**.  In the search box on the next screen, enter **react**.  Select **ASP.NET Core with React.js** from the list of templates and then **Next**.  

This **React.js** template specifies how to integrate React.js with an ASP.NET Core app.  

> :::image type="complex" source="./media/create-new-project.png" alt-text="Create a new ASP.NET Core Web app with React.js" lightbox="./media/create-new-project.png":::

Reviewer: If not too much trouble/time, in screenshots, change yellow box to red.  Usually 3px not 5px, but that width could be hard to change.  
To change yellow to red in Paint:
1. Shift+click an image link in the .md file in VS Code.
2. Use a menu item or Sh+Alt+R to open Explorer at the dir.
3. Right-click an image file, Edit, to open in Paint.
4. Select or enter a red color.  Or open an image with red box and use Dropper tool to sample it.
4. Select Paint Bucket tool, click a yellow rectangle.  Save.

   Create a new ASP.NET Core Web app with React.js  
:::image-end:::  

### Launch Microsoft Edge from Visual Studio  

> After you create your project, open `ClientApp/src/components/Counter.js`.  Now, to use Visual Studio to debug JavaScript, select the dropdown next to the green **Play** button and **IIS Express**.  

Reviewer: ```suggestion
1.  After you create your project, open `ClientApp/src/components/Counter.js`.

1.  Now, to use Visual Studio to debug JavaScript, select the dropdown next to the green **Play** button and **IIS Express**.  
```
For closely related (conceptually grouped) user-actions that feel like a single conceptual step, group into a single numbered list item.  Avoid ultra-granular, fragmented numbered steps.
blank line spreads numbered steps apart, for readability.  
For the image block below, prefix spaces to move it right so it starts in column 4.  I think most Markdown, including this build system, treats column 4 as meaningful/significant; have to check the Doc Contributor Guide later.

> 

Reviewer: near top of article, change:
Visual Studio completes the following workflow using a single button.
to:
Visual Studio completes the entire following workflow when you select a single button.
(which button? does the surrounding text fill-in a mental picture adequately to follow the meaning here?  i'd try to slip-in a hint of what we have in mind)

Author: This section wasn't very clear, I rephrased it.

> :::image type="complex" source="./media/vs-dropdown.png" alt-text="The dropdown next to the green Play button and IIS Express" lightbox="./media/vs-dropdown.png":::
>    The dropdown next to the green **Play** button and **IIS Express**  
> :::image-end:::  

Reviewer: ```suggestion
    :::image type="complex" source="./media/vs-dropdown.png" alt-text="The dropdown next to the green Play button and IIS Express" lightbox="./media/vs-dropdown.png":::
       The dropdown next to the green **Play** button and **IIS Express**  
    :::image-end:::  
```
Start in col 4(?) to indent image and maybe to make the numbered steps below auto-increment correctly.

Select **Script Debugging** > **Enabled**.  

:::image type="complex" source="./media/enable-script-debugging.png" alt-text="Turn on script debugging in Visual Studio" lightbox="./media/enable-script-debugging.png":::
   Turn on script debugging in Visual Studio  
:::image-end:::  

> In the same dropdown, select **Web Browser** > the preview channel of Microsoft Edge that you want Visual Studio to launch, such as Microsoft Edge Canary, Dev, or Beta.  If you are not already using one of the Microsoft Edge preview channels, navigate to [Download Microsoft Edge Insider Channels][MicrosoftedgeinsiderDownload] to download one.  

Reviewer: Elsewhere, change:
you may step through the script line-by-line
to:
you can step through the script line-by-line

:::image type="complex" source="./media/set-web-browser.png" alt-text="Select the preview channel of Microsoft Edge that you want Visual Studio to launch" lightbox="./media/set-web-browser.png":::
   Select the preview channel of Microsoft Edge that you want Visual Studio to launch  
:::image-end:::  

Now that Visual Studio is correctly configured, select the green **Play** button.  Visual Studio builds your app, starts the web server, launches Microsoft Edge, and navigates to `https://localhost:44362/` or whatever port is specified in `launchSettings.json`.  

:::image type="complex" source="./media/edge-launch.png" alt-text="Microsoft Edge launches from Visual Studio" lightbox="./media/edge-launch.png":::
   Microsoft Edge launches from Visual Studio  
:::image-end:::  

### Debug JavaScript running in Microsoft Edge  

> Switch back to Visual Studio.  In `Counter.js`, to set a breakpoint on Line 13, select the gutter next to the line.  

Reviewer: Depending on how thorough of an update this PR aims for, test the steps, make sure it's still Line 13.  Testing takes a long time, but can yield corrections.  But consider/balance/weigh the opportunity cost of such in-depth testing of steps.

Author: Yep, I did test locally, which is how I was able to refresh a couple of the screenshots. The line is still 13, so this is correct.

:::image type="complex" source="./media/set-breakpoint.png" alt-text="Select the gutter next to Line 13 in Counter.js to set a breakpoint in Visual Studio" lightbox="./media/set-breakpoint.png":::
   Select the gutter next to Line 13 in `Counter.js` to set a breakpoint in Visual Studio  
:::image-end:::  

> Now switch back to the instance of Microsoft Edge that Visual Studio launched.  Select **Counter** in the navigation menu on the left of the webpage.  Now select **Increment**.  

Reviewer: ```suggestion
1.  Switch back to the instance of Microsoft Edge that Visual Studio launched.

1.  Select **Counter** in the navigation menu at the top of the webpage, and then select the **Increment** button.
```
Check the location of "Counter" UI, in context.    Looks like it's in the upper right in the screenshot.
Style guide says "don't state the UI control type, unless it helps clarity".  My retort is, it _always_ helps clarity, esp given the Accessibility rule, "don't assume the screenshots are visible to the audience".  I omit "the ... button" only in the easy case of the OK button.

## Attach to Microsoft Edge  

> Previously, you launched Microsoft Edge from Visual Studio.  You can also attach the Visual Studio debugger to an already running instance of Microsoft Edge.  

Author: Yeah, 2 alternative approaches that can't be used at the same time. I'll update accordingly.

First, ensure that there are no running instances of Microsoft Edge.  Now, from your command line, run the following command.  

```console
start msedge –remote-debugging-port=9222
```  

From Visual Studio, select **Debug** > **Attach to Process** or `Ctrl`+`Alt`+`P`.  

:::image type="complex" source="./media/attach-to-process.png" alt-text="Select Attach to Process in Visual Studio" lightbox="./media/attach-to-process.png":::
   Select **Attach to Process** in Visual Studio  
:::image-end:::  

> From the **Attach to Process** dialog, set **Connection type** to **Chrome devtools protocol websocket (no authentication)**.  In the **Connecting target** textbox, type `http://localhost:9222/` and select `Enter`.  Review the list of open tabs you have in Microsoft Edge listed out in the **Attach to Process** dialog.  

Reviewer: ```suggestion
1.  In the **Attach to Process** dialog, set **Connection type** to **Chrome devtools protocol websocket (no authentication)**.  
1.  In the **Connecting target** textbox, type `http://localhost:9222/` and select `Enter`.  In the **Attach to Process** dialog, review the list of tabs that are open in Microsoft Edge.  
```
Break into 2-3 cohesive steps.  Reading this step out of context, it's unclear how many dialog boxes are involved.  Each dialog would be a separate step.

   Configure the **Attach to Process** dialog in Visual Studio  
:::image-end:::  

> Select **Select** > the checkbox next to **JavaScript (Microsoft Edge – Chromium)**.  To add tabs, navigate to new tabs, and close tabs and display the changes reflected in the **Attach to Process** dialog, select the **Refresh** button.  Choose the tab you want to debug and select **Attach**.  

Author: Thanks for catching this. I was mostly focusing on the first part of the page (`Debug JavaScript running in Microsoft Edge`) and didn't realize that (`Attach to Microsoft Edge`) was incorrect. Turns out this whole section was very confusing as it was talking about VS and Edge at the same time but didn't make it clear. I've tested the workflow again myself and will update accordingly.

> The Visual Studio debugger is now attached to Microsoft Edge.  You may pause the running of JavaScript, set breakpoints, and review `console.log()` statements directly in the Debug Output window in Visual Studio.  

Reviewer: Global: check 'can' and 'may'.  Change 'may' to 'can' in most places.  Per MWSG's [can, may](https://styleguides.azurewebsites.net/Styleguide/Read?id=2700&topicid=33430) article: "When ability is what you need to express, it's OK to use can to describe actions or tasks that the reader or program is able to do. Use might to express possibility. **Don't use may**, which might be interpreted as providing permission."

> The Visual Studio debugger is now attached to Microsoft Edge.  You may pause the running of JavaScript, set breakpoints, and review `console.log()` statements directly in the Debug Output window in Visual Studio.  

Reviewer: ```suggestion
The Visual Studio debugger is now attached to Microsoft Edge.  You can pause the running of JavaScript, set breakpoints, and review `console.log()` statements directly in the Debug Output window in Visual Studio.  
```
I'd keep this left-justified.  Optionally shift w/ leading spaces to start in column 4(?) if it looks more consistent.
